### PR TITLE
Lmb 796 export mandatarissen as csv

### PR DESCRIPTION
**Pull request by JonasVanHoof**
## Description

We want the mandatarissen to be downloadable. The file includes the required columns and the data will be filtered the way it was in the frontend. Creating this export is done in the mandataris service

## How to test

Open the mandataris search page or the organen>mandatarissen page and download the content. See if the results have the active filters applied. You will will also see previous mandatarissen of the person but this is how we want it.

## Links to other PR's

- https://github.com/lblod/mandataris-service/pull/55

## Attachments

image

### CODE REVIEW

File: app/services/mandataris-api.js [hunk 1]
```diff
@@ -104,4 +106,66 @@ export default class MandatarisApiService extends Service {
       include: 'fractietype',
     });
   }
+
+  async downloadAsCsv({
```
Comment (Rahien) [inline app/services/mandataris-api.js:110]: why not just use an html-5 download link? this is the old way of doing things from before this existed but browser support seems good now

File: app/services/mandataris-api.js [hunk 2]
```diff
@@ -104,4 +106,66 @@ export default class MandatarisApiService extends Service {
...
+    fractieIds = [],
+    hasFilterOnOnafhankelijkeFractie = false,
+    hasFilterOnNietBeschikbareFractie = false,
+    bestuursFunctieCodeIds = [],
+    sort = null,
+  }) {
+    const response = await fetch(
+      `${API.MANDATARIS_SERVICE}/mandatarissen/download`,
+      {
+        method: 'POST',
```
Comment (Rahien) [inline app/services/mandataris-api.js:124]: feels weird that this is a post and not a get

File: app/services/mandataris-api.js (hunk 1 above)
Comment (JonasVanHoof) [reply app/services/mandataris-api.js:110]: Can you add a json body to this?

File: app/components/download-mandatarissen-from-table.hbs [hunk 3]
```diff
@@ -0,0 +1,22 @@
+<Shared::Tooltip
+  @showTooltip={{true}}
+  @tooltipText="Exporteer mandatarissen naar CSV formaat"
+>
+  <AuLinkExternal
+    {{did-update
```
Comment (JonasVanHoof) [inline app/components/download-mandatarissen-from-table.hbs:6]: Necessary to update the download link when a filter changes

File: app/services/mandataris-api.js [hunk 4]
```diff
@@ -104,4 +104,30 @@ export default class MandatarisApiService extends Service {
...
+    bestuursorgaanId = null,
+    activeOnly = false,
+    persoonIds = null,
+    fractieIds = null,
+    hasFilterOnOnafhankelijkeFractie = false,
+    hasFilterOnNietBeschikbareFractie = false,
+    bestuursFunctieCodeIds = null,
+    sort = null,
+  }) {
+    const queryParams = [
```
Comment (JonasVanHoof) [inline app/services/mandataris-api.js:119]: Rather do  it here than in the download component

File: app/components/download-mandatarissen-from-table.js [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+      persoonIds: this.persoonIds,
+      fractieIds: this.fractieIds,
+      hasFilterOnOnafhankelijkeFractie: this.filterOnOnafhankelijkeFractie,
+      hasFilterOnNietBeschikbareFractie: this.filterOnNietBeschikbaarFractie,
+      bestuursFunctieCodeIds: this.bestuursFunctieCodeIds ?? [],
+      sort: this.args.sort,
+    });
+  });
+
+  async getBestuursorgaanInTijdForPeriod() {
```
Comment (JonasVanHoof) [inline app/components/download-mandatarissen-from-table.js:41]: move one component up and pass as argument to the component

**Review by Rahien — APPROVED** [review]

**Pull request merged**